### PR TITLE
uint: implement integer_sqrt

### LINKED ISSUE #551: implement num_traits::int::PrimInt for U256 / U512
This allow U256 / U512 to be used with crates like `integer-sqrt`

### PR DESCRIPTION
**Pull request by ordian**
Closes #551.

~~The algorithm implemented is the same as in the `integer-sqrt` crate and it's quite slow, on my machine u256 isqrt takes 2μs (>2000ns) and u512 > 5μs.~~

~~Do you know of any better algorithms?~~

### CODE REVIEW

**Comment by ordian** [comment]
@xlc are you OK with `integer-sqrt` support, or do you actually need `num_traits::int::PrimInt` impl?

**Comment by xlc** [comment]
`integer-sqrt` will do it for now.

File: primitive-types/impls/num-traits/Cargo.toml [hunk 1]
```diff
@@ -9,4 +9,5 @@ edition = "2018"
 
 [dependencies]
 num-traits = { version = "0.2", default-features = false }
+integer-sqrt = "0.1"
```
[the remark this answers is not in the document]
Comment (ordian) [reply primitive-types/impls/num-traits/Cargo.toml:12]: Technically it is, but it's rather not useful by its own. E.g. see https://github.com/paritytech/substrate/blob/d8b3fce28513169d8ee1315dd9a4cd92d95753cc/primitives/arithmetic/src/traits.rs#L33-L51

File: uint/fuzz/fuzz_targets/div_mod.rs [hunk 2]
```diff
@@ -32,5 +32,5 @@ fuzz_target!(|data: &[u8]| {
 			let (a, b) = x_gmp.div_rem(y_gmp);
 			assert_eq!((from_gmp(a), from_gmp(b)), x.div_mod(y));
 		}
-    }
+	}
```
[the remark this answers is not in the document]
Comment (ordian) [reply uint/fuzz/fuzz_targets/div_mod.rs:35]: It's GH, not me :shrug:

**Review by dvdplm — APPROVED** [review]
LGTM but I also have not looked deeply into the algos used here.

**Review by sorpaas — APPROVED** [review]

**Pull request merged**